ext/openssl/ossl_asn1.c

```diff
@@ -1818,13 +1842,14 @@ do{\
     rb_define_method(cASN1ObjectId, "oid", ossl_asn1obj_get_oid, 0);
     rb_define_alias(cASN1ObjectId, "short_name", "sn");
     rb_define_alias(cASN1ObjectId, "long_name", "ln");
+    rb_define_method(cASN1ObjectId, "==", ossl_asn1obj_eq, 1);
     rb_attr(cASN1BitString, rb_intern("unused_bits"), 1, 1, 0);
 
     rb_define_method(cASN1EndOfContent, "initialize", ossl_asn1eoc_initialize, 0);
     rb_define_method(cASN1EndOfContent, "to_der", ossl_asn1eoc_to_der, 0);
 
     class_tag_map = rb_hash_new();
-    rb_gc_register_mark_object(class_tag_map);
+    rb_global_variable(&class_tag_map);
```

Comment:
This reverts the fix for https://bugs.ruby-lang.org/issues/16196, and will cause a segfault.

Reply:
Okay so it was never upstreamed? I can pull this into OpenSSL source code. Should we check for other issues like this?

Reply:
There may be other fixes to be ported. 😭

Reply:
Okay, I will upstream all these changes.